sensors/dependencies/filter.go

```diff
@@ -252,3 +256,60 @@ filter:
 	}
 	return true, nil
 }
+
+// filterExpr applies expression based filters against event data
+func filterExpr(filters []v1alpha1.ExprFilter, event *v1alpha1.Event) (bool, error) {
+	if filters == nil {
+		return true, nil
+	}
+	if event == nil {
+		return false, fmt.Errorf("nil event")
+	}
+	payload := event.Data
+	if payload == nil {
+		return true, nil
+	}
+	var js *json.RawMessage
+	if err := json.Unmarshal(payload, &js); err != nil {
+		return false, err
+	}
+	var jsData []byte
+	jsData, err := json.Marshal(js)
+	if err != nil {
+		return false, err
+	}
+
+	for _, filter := range filters {
+		parameters := map[string]interface{}{}
+		for _, field := range filter.Fields {
+			result := gjson.GetBytes(jsData, field.Path)
+			if !result.Exists() {
+				return false, fmt.Errorf("path %s does not exist", field.Path)
+			}
+			switch field.Type {
+			case v1alpha1.JSONTypeString:
+				parameters[field.Name] = result.Str
```

Comment:
Can you try `parameters[field.Name] = result.Value()`? I feel like that will work, and `fields` is not required.

Reply:
yeah that would work